spec/frameSpec.js

```diff
@@ -0,0 +1,169 @@
+describe("Frame", function() {
+  var frame, finalFrame, testMultipliers, firstRoll;
+  beforeEach(function() {
+    testMultipliers = [2, 1];
+    frame = new Frame(testMultipliers, false);
+    finalFrame = new Frame(testMultipliers, true);
+    firstRoll = 2;
+  });
+
+  it("is initialised with an empty rolls array", function() {
+    expect(frame.rolls).toEqual([]);
+  });
+
+  it("is initialised with the passed multiplier array", function() {
+    expect(frame.multipliers).toEqual([2,1]);
+  });
+
+  it("is initialised with score 0", function() {
+    expect(frame.score).toEqual(0);
+  });
+
+  it("is inititialised with isFinalFrame", function() {
+    expect(frame.isFinalFrame).toEqual(false);
+    expect(finalFrame.isFinalFrame).toEqual(true);
+  });
+
+  describe(".roll()", function() {
+    it("stores the roll in this.rolls", function() {
+      frame.roll(firstRoll);
+
+      expect(frame.rolls).toEqual([firstRoll]);
+    });
+
+    it("calls this._updateScore()", function() {
+      spyOn(frame, "_updateScore");
+      frame.roll(firstRoll);
+
+      expect(frame._updateScore).toHaveBeenCalledWith(firstRoll);
+
+    });
+
+    it("calls this._calculateMultipliers", function() {
+      spyOn(frame, "_calculateMultipliers");
+      frame.roll(firstRoll);
+
+      expect(frame._calculateMultipliers).toHaveBeenCalledWith();
+    });
+  });
+
+  describe(".updateScore()", function() {
+    it("calculates the score and adds it to this.score", function() {
+      let currentMultiplier = testMultipliers[0];
+      frame._updateScore(firstRoll);
+
+      expect(frame.score).toEqual(currentMultiplier * firstRoll);
+    });
+  });
+
+  describe("._calculateMultipliers", function() {
+    let secondRoll, strikeRoll, spareRoll;
+    beforeEach(function() {
+      secondRoll = 6;
+      spareRoll = 8;
+      strikeRoll = 10;
+      frame.multipliers = [1];
+    });
+
+    it("when final frame, leaves first multiplier untouched, sets second to 1", function() {
+      finalFrame.multipliers = [1];
+      finalFrame.rolls = [strikeRoll];
+      finalFrame._calculateMultipliers();
+
+      expect(finalFrame.multipliers).toEqual([1,1]);
```

Comment:
I think I've worked out what 'multipliers' is... (I haven't looked at the implementation yet, just the tests). I think this is related to bonuses. When it comes to naming and implementation, it's useful to think about whether the purposes of a method or property would be obvious to somebody who was coming into the code from the cold. I might be tempted to call this 'bonusMultiplier' or something along those lines, to make it clear what the context of the multiplier is.